libfreerdp/core/tcp.c

```diff
@@ -222,12 +294,81 @@ static long transport_bio_simple_ctrl(BIO* bio, int cmd, long arg1, void* arg2)
 	return status;
 }
 
+static int transport_bio_simple_init(BIO* bio, SOCKET socket, int shutdown)
+{
+	WINPR_BIO_SIMPLE_SOCKET* ptr = (WINPR_BIO_SIMPLE_SOCKET*) bio->ptr;
+
+	ptr->socket = socket;
+
+	bio->shutdown = shutdown;
+	bio->flags = BIO_FLAGS_SHOULD_RETRY;
+	bio->init = 1;
+
+	if (ptr->win32)
```

Comment:
yeah... it reduced the amount of #ifdefs necessary, and we're crazy enough to have all the functions in WinPR anyway. I don't strongly care about having it runtime, it just made the code slightly cleaner...